firefox-ios/Client/Frontend/Browser/BrowserViewController/Extensions/BrowserViewController+DownloadQueueDelegate.swift

```diff
@@ -5,51 +5,94 @@
 import Common
 import Foundation
 
-extension BrowserViewController: DownloadQueueDelegate {
+extension BrowserViewController: DownloadQueueDelegate, DownloadCancellationDelegate {
     func downloadQueue(_ downloadQueue: DownloadQueue, didStartDownload download: Download) {
         // For now, each window handles its downloads independently; ignore any messages for other windows' downloads.
         let uuid = windowUUID
         guard download.originWindow == uuid else { return }
-
-        // If no other download toast is shown, create a new download toast and show it.
-        guard let downloadToast = self.downloadToast else {
-            presentDownloadProgressToast(download: download, windowUUID: uuid)
+        if let downloadProgressManager = self.downloadProgressManager {
```

Comment:
```suggestion

        if let downloadProgressManager = self.downloadProgressManager {
```
Let's leave the extra white space to separate from the guard check.